go/cmd/dolt/commands/stashcmds/stash.go

```diff
@@ -137,17 +139,22 @@ func stashChanges(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgPars
 	if err != nil {
 		return fmt.Errorf("couldn't get working root, cause: %s", err.Error())
 	}
-
 	if headHash.Equal(stagedHash) {
 		if headHash.Equal(workingHash) {
 			cli.Println("No local changes to save")
 			return nil
-		} else if allUntracked, err := allAreUntrackedFilesInWorkingSet(ctx, roots); err != nil {
+		} else if allUntracked, err := workingSetContainsOnlyUntrackedTables(ctx, roots); err != nil {
 			return err
 		} else if !apr.Contains(IncludeUntrackedFlag) && allUntracked {
 			// if all changes in working set are untracked files, then no local changes to save
 			cli.Println("No local changes to save")
 			return nil
+		} else if allIgnored, err := workingSetContainsOnlyIgnoredTables(ctx, roots); err != nil {
+			return err
+		} else if !apr.Contains(AllFlag) && allIgnored {
+			// if all changes in working set are ignored files, then no local changes to save
+			cli.Println("No local changes to save")
+			return nil
```

Comment:
Reworked this function because it wasn't even correct.

Only untracked tables can be ignored (this is consistent with git), and whatever documentation we release for dolt_ignore should mention this. So in your example case, the ignored table will be stashed only when --all is passed, and the untracked table will be stashed when either -u or --all is passed. Tests have been added for this.